utils/fixed_decimal/src/signed_decimal.rs

```diff
@@ -0,0 +1,3364 @@
+    ///
+    /// let decimal =
+    ///     SignedFixedDecimal::try_from_f64(-5.1, FloatPrecision::Magnitude(-2))
+    ///         .expect("Finite quantity with limited precision");
+    /// assert_writeable_eq!(decimal, "-5.10");
+    ///
+    /// let decimal =
+    ///     SignedFixedDecimal::try_from_f64(0.012345678, FloatPrecision::RoundTrip)
+    ///         .expect("Finite quantity");
+    /// assert_writeable_eq!(decimal, "0.012345678");
+    ///
+    /// let decimal =
+    ///     SignedFixedDecimal::try_from_f64(12345678000., FloatPrecision::Integer)
+    ///         .expect("Finite, integer-valued quantity");
+    /// assert_writeable_eq!(decimal, "12345678000");
+    /// ```
+    ///
+    /// Negative zero is supported.
+    ///
+    /// ```rust
+    /// use fixed_decimal::{SignedFixedDecimal, FloatPrecision};
+    /// use writeable::assert_writeable_eq;
+    ///
+    /// // IEEE 754 for floating point defines the sign bit separate
+    /// // from the mantissa and exponent, allowing for -0.
+    /// let negative_zero =
+    ///     SignedFixedDecimal::try_from_f64(-0.0, FloatPrecision::Integer)
+    ///         .expect("Negative zero");
+    /// assert_writeable_eq!(negative_zero, "-0");
+    /// ```
+    pub fn try_from_f64(float: f64, precision: FloatPrecision) -> Result<Self, LimitError> {
+        match float.is_sign_negative() {
+            true => Ok(SignedFixedDecimal {
+                sign: Sign::Negative,
+                absolute: UnsignedFixedDecimal::try_from_f64(-float, precision)?,
+            }),
+            false => Ok(SignedFixedDecimal {
+                sign: Sign::None,
+                absolute: UnsignedFixedDecimal::try_from_f64(float, precision)?,
+            }),
+        }
+    }
+}
+
+/// All the rounding and rounding related logic is implmented in this implmentation block.
+impl SignedFixedDecimal {
+    /// Rounds this number at a particular digit position.
+    ///
+    /// This uses half to even rounding, which rounds to the nearest integer and resolves ties by
+    /// selecting the nearest even integer to the original value.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use fixed_decimal::SignedFixedDecimal;
+    /// # use std::str::FromStr;
+    ///
+    /// let mut dec = SignedFixedDecimal::from_str("-1.5").unwrap();
+    /// dec.round(0);
+    /// assert_eq!("-2", dec.to_string());
+    /// let mut dec = SignedFixedDecimal::from_str("0.4").unwrap();
+    /// dec.round(0);
+    /// assert_eq!("0", dec.to_string());
+    /// let mut dec = SignedFixedDecimal::from_str("0.5").unwrap();
+    /// dec.round(0);
+    /// assert_eq!("0", dec.to_string());
+    /// let mut dec = SignedFixedDecimal::from_str("0.6").unwrap();
+    /// dec.round(0);
+    /// assert_eq!("1", dec.to_string());
+    /// let mut dec = SignedFixedDecimal::from_str("1.5").unwrap();
+    /// dec.round(0);
+    /// assert_eq!("2", dec.to_string());
+    /// ```
+    pub fn round(&mut self, position: i16) {
```

Comment:
Thought: Rounding modes are not consistently applicable to signed vs unsigned:

| Rounding Mode | Signed? | Unsigned? |
|---|---|---|
| (Half)Trunc | Yes | Yes |
| (Half)Expand | Yes | Yes |
| (Half)Ceil | Yes | No |
| (Half)Floor | Yes | No |
| HalfEven | Yes | Yes |

Maybe define the ones that work the same on the unsigned type on only the unsigned type and the ones that only work on the signed type to be on the signed type?